### DISCUSSION TITLE
Tasmota Scripting into Berry Basics

### DISCUSSION THREAD

**Discussion by mmmmm009** [arendst/Tasmota#20825] [Q&A] [answered]
Hi,

I'd like to "translate" the following Tasmota script to Berry:
I would appreciate your help.

->D
val=0
->T
val=adc(2)

if val<512
then
print Below 512  
endif

if val>512
then 
print Above 512  
endif

**Comment by sfromis** [comment, marked as answer]
I'm not familiar with all intricacies of scripting, but probably something similar to this:
```
tasmota.add_rule('ANALOG#A1',
  def (value,trigger,data)
    if value < 512
      print('less than')
    elif value > 512
      print('greater than')
    else
      print('equals')
    end
  end)
```
This needs you to config the relevant pin as analog, probably "ADC Input".

For repeated testing with modified code, first `tasmota.remove_rule('ANALOG#A1')` to avoid the old one lingering.
